client/asset/eth/rpcclient.go

```diff
@@ -256,6 +262,64 @@ func (c *rpcclient) initiate(txOpts *bind.TransactOpts, netID int64, refundTimes
 	return c.es.Initiate(txOpts, big.NewInt(refundTimestamp), secretHash, *participant)
 }
 
+// initiateGas checks the amount of gas that is used for a call to the initiate function.
+func (c *rpcclient) initiateGas(ctx context.Context, contractAddress *common.Address) (uint64, error) {
+	parsedAbi, err := abi.JSON(strings.NewReader(swap.ETHSwapABI))
+	if err != nil {
+		return 0, err
+	}
+	refundTimestamp := time.Now().Unix()
+	var secret [32]byte
+	copy(secret[:], encode.RandomBytes(32))
+	secretHash := sha256.Sum256(secret[:])
+	// just an arbitrary address
+	participant := common.HexToAddress("345853e21b1d475582E71cC269124eD5e2dD3422")
+	data, err := parsedAbi.Pack("initiate", big.NewInt(refundTimestamp), secretHash, participant)
```

Comment:
> It's a preloaded simnet address. 

I don't think addresses have any kind of encoding or checksum, I think they're just 20 bytes as well, the hash of the pubkey.

Reply:
Maybe that gas error is because we are using with the harness and that address in the test actually does have funds. That would be a problem with making these functions usable for every start up to get estimates, if you actually need funds to make a swap to get the estimate.

Reply:
There needs to be enough gas to pay for the value of the swap, so an account with an empty balance will not work. The gas price is set to 0, so that doesn't matter.